tests/src/application/middlewares/auth.spec.ts

```diff
@@ -35,111 +35,106 @@ const makeSut = (): SutTypes => {
       const httpResponse = await sut.handle(httpRequest)
-
-      expect(httpResponse).toEqual(unauthorized())
+      expect(httpResponse).toEqual(unauthorized(new InvalidTokenError('Invalid or expired token')))
     })
+  })
 
-    it('Should call loadGuardianById with correct value', async () => {
-      const { sut, loadGuardianByIdStub } = makeSut()
-      const spyLoadGuardianById = jest.spyOn(loadGuardianByIdStub, 'loadById')
-
-      const httpRequest = makeFakeAuthorization({ data: 'valid_token' })
+  describe('LoadByGuardianById', () => {
+    it('Should call LoadByGuardianById with correct userId', async () => {
+      const { sut, loadGuardianByIdStub, tokenDecoderStub } = makeSut()
+      const loadByIdSpy = jest.spyOn(loadGuardianByIdStub, 'loadById')
+      const sub = 'any_id'
+      jest.spyOn(tokenDecoderStub, 'decode').mockResolvedValue({ sub })
       await sut.handle(httpRequest)
-
-      expect(spyLoadGuardianById).toHaveBeenCalledWith(makeFakePayload().sub)
+      expect(loadByIdSpy).toHaveBeenCalledWith(sub)
     })
 
-    it('Should return 401 if authorization not match with accessToken in database', async () => {
-      const { sut, hashComparerStub } = makeSut()
-      jest.spyOn(hashComparerStub, 'compare').mockResolvedValueOnce(false)
-
-      const httpRequest = makeFakeAuthorization({ data: 'valid_token' })
+    it('Should return 500 (ServerError) if LoadByGuardianById throws', async () => {
+      const { sut, loadGuardianByIdStub } = makeSut()
+      jest.spyOn(loadGuardianByIdStub, 'loadById').mockRejectedValue(new Error())
       const httpResponse = await sut.handle(httpRequest)
-
-      expect(httpResponse).toEqual(unauthorized())
+      expect(httpResponse).toEqual(makeFakeServerError())
     })
 
-    it('Should return 500 if loadGuardianById throws', async () => {
+    it('Should return 401 (Unauthorized) if hash and value do not match', async () => {
       const { sut, loadGuardianByIdStub } = makeSut()
-      jest.spyOn(loadGuardianByIdStub, 'loadById').mockRejectedValueOnce(new Error())
-
-      const httpRequest = makeFakeAuthorization({ data: 'any_token' })
+      jest.spyOn(loadGuardianByIdStub, 'loadById').mockResolvedValue(undefined)
       const httpResponse = await sut.handle(httpRequest)
-
-      expect(httpResponse).toEqual(makeFakeServerError())
+      expect(httpResponse).toEqual(unauthorized(new NotFoundError('User not found')))
     })
   })
 
-  describe('test authMiddleware success case', () => {
-    it('Should return 200 if success', async () => {
-      const { sut } = makeSut()
-      const httpRequest = makeFakeAuthorization({ data: 'valid_token' })
+  test('Should return 400 (Unauthorized) if authorization is not provided', async () => {
+    const { sut } = makeSut()
+    const httpRequests = [
+      { authorization: '' },
+      { authorization: null },
+      { authorization: undefined }
+    ]
+    for (const request of httpRequests) {
```

Comment:
Very good, we would like adopt this strategy to another cases, like test unhandler erros by dependencies. 
Jest have this function to parametrize tests. See the reference:
`https://jestjs.io/pt-BR/blog/2018/05/29/jest-23-blazing-fast-delightful-testing#jest-each`